source/driver-vs-mongoose.txt

```diff
@@ -0,0 +1,142 @@
+manner. 
+
+Differences in Schema Management
+--------------------------------
+
+Both the {+driver-long+} and Mongoose let you apply schema validation
+to your collection. However, the biggest difference between the two are
+the way they each set and manage schema validation.
+
+Mongoose allows you to define data schemas using its schema notation.
+These schemas define the structure of the documents to be stored in
+MongoDB, along with data types, default values, and validation rules.
+Mongoose provides built-in support for data validation based on the
+defined schema rules. It ensures that the data being saved to the
+database adheres to the specified schema, preventing invalid data from
+being stored. 
+
+The {+driver-long+} applies the schema validation at the database
+level. In comparison, Mongoose applies schema validation at the application level. In the
+{+driver-short+} schema validation works on individual collections by
+using either `JSON Schema Validation <https://www.mongodb.com/docs/manual/core/schema-validation/specify-json-schema/>`__
+or `Schema Validation with Query Operators <https://www.mongodb.com/docs/manual/core/schema-validation/specify-query-expression-rules/>`__.
+Validation with query operators is useful for dynamic validation rules
+calculated at runtime. By contrast, JSON validation is useful for, among other
+reasons, being an open standard which makes it easy to connect to
+third-party libraries.
+
+Similarities
+------------
+
+As Mongoose is built upon the {+driver-long+}, they share many
+capabilities. Even if you use Mongoose, you always have the option to
+use any native function or command built into the {+driver-short+}
+alongside Mongoose. Mongoose also offers its own methods that are quite
+similar to the {+driver-short+}, such as ``updateOne()`` and
+``replaceOne()``.
+
+.. tip::
+   
+   There are many other Mongoose specific methods and APIs not discussed. For
+   more information on how to use Mongoose, check out the official `Mongoose
+   documentation <https://mongoosejs.com/docs/>`__.
+
+Additional Functionality Provided by Mongoose
+---------------------------------------------
+
+On top of schema management, there is a multitude of additional
+functionality provided by Mongoose. In this section, we will go over a
+few of the additional functionalities.
+
+Data Modeling
+~~~~~~~~~~~~~
+
+Mongoose introduces the concept of models, which are JavaScript classes
+defined based on schemas. Models act as an interface to interact
+with MongoDB collections, abstracting away the complexities of database
+operations.
+
+Middleware Hooks
+~~~~~~~~~~~~~~~~
+
+Mongoose supports middleware hooks, both pre and post-hooks, that allow you
+to add custom logic before or after certain database operations like
+save, update, and remove. This is useful for implementing additional
+actions or validations. Mongoose supports adding custom plugins and
+middleware functions to extend its functionality further. 
```

Comment:
Q: similar to above question -- what actions or validations? What are these custom plugins/functions for?

Reply:
Similar answer to the above question. There are a multitude of actions or validations available for use with middleware and the custom plugins are outside the scope of Mongoose's official offerings so I didn't think to leave an example. I think it would be better to link to the Mongoose documentation on middleware so the user can see all the uses if they want to learn more but am open to other opinions.